unisave-rest/src/test/java/org/uniprot/api/unisave/controller/UniSaveControllerTest.java

```diff
@@ -153,6 +157,228 @@ void canRetrieveSpecificVersionsOfEntriesWithContent() throws Exception {
+                .andExpect(
+                        header().string(
+                                        HttpHeaders.CONTENT_TYPE,
+                                        UniProtMediaType.FASTA_MEDIA_TYPE_VALUE))
+                .andExpect(
+                        content()
+                                .string(
+                                        containsString(
+                                                ">P12345: EV=3-4 SV=2\n"
+                                                        + "MASGAYSKYLFQIIGETVSSTNRGNKYNSFDHSRVDTRAGSFREAYNSKKKGSGRFGRKC\n"
+                                                        + "FQIIGETVSSTNRG\n"
+                                                        + ">P12345: EV=1-2 SV=1\n"
+                                                        + "MASGAYSKYLFQIIGETVSSTNRGNKYNSFDHSRVDTRAGSFREAYNSKKKGSGRFGRKC\n"
+                                                        + "FQIIGETVSSTNRG")));
+    }
+
+    @Test
+    void canRetrieveAggregatedEntriesInVersionRange() throws Exception {
+        // given
+        when(uniSaveRepository.retrieveEntry(ACCESSION, 4)).thenReturn(mockEntry(ACCESSION, 4, 2, true));
+        when(uniSaveRepository.retrieveEntry(ACCESSION, 3)).thenReturn(mockEntry(ACCESSION, 3, 2, true));
+        when(uniSaveRepository.retrieveEntry(ACCESSION, 2)).thenReturn(mockEntry(ACCESSION, 2, 1, true));
+        when(uniSaveRepository.retrieveEntry(ACCESSION, 1)).thenReturn(mockEntry(ACCESSION, 1, 1, true));
+
+        // when
+        ResultActions response =
+                mockMvc.perform(
+                        get(RESOURCE_BASE + ACCESSION)
+                                .header(ACCEPT, UniProtMediaType.FASTA_MEDIA_TYPE_VALUE)
+                                .param("uniqueSequences", "true")
+                                .param("versions", "2-4"));
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.OK.value()))
+                .andExpect(
+                        header().string(
+                                HttpHeaders.CONTENT_TYPE,
+                                UniProtMediaType.FASTA_MEDIA_TYPE_VALUE))
+                .andExpect(
+                        content()
+                                .string(
+                                        containsString(
+                                                ">P12345: EV=3-4 SV=2\n"
+                                                        + "MASGAYSKYLFQIIGETVSSTNRGNKYNSFDHSRVDTRAGSFREAYNSKKKGSGRFGRKC\n"
+                                                        + "FQIIGETVSSTNRG\n"
+                                                        + ">P12345: EV=2 SV=1\n"
+                                                        + "MASGAYSKYLFQIIGETVSSTNRGNKYNSFDHSRVDTRAGSFREAYNSKKKGSGRFGRKC\n"
+                                                        + "FQIIGETVSSTNRG")));
+    }
+
+    @Test
+    void canRetrieveAggregatedEntriesForCSV() throws Exception {
```

Comment:
Suggestion: CSVVersions?
Do you think that adding version at the end will help to understand?
if Yes, can you please apply it in other places as well?

Reply:
yep and done